main/src/cgeo/geocaching/ui/CacheDetailsCreator.java

```diff
@@ -227,4 +230,65 @@ public TextView addHiddenDate(@NonNull final Geocache cache) {
         view.setId(R.id.date);
         return view;
     }
+
+    public void addLatestLogs(final Geocache cache) {
+        Context context = parentView.getContext();
+
+        final RelativeLayout layout = (RelativeLayout) activity.getLayoutInflater().inflate(R.layout.cache_information_item, null, false);
+        final TextView nameView = layout.findViewById(R.id.name);
+        nameView.setText(res.getString(R.string.cache_latest_logs));
+        final LinearLayout markers = layout.findViewById(R.id.linearlayout);
+
+        int smileySize = 48;
+        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(smileySize, smileySize);
+        lp.setMargins(0, 0, 5, 0);
+
+        List<LogEntry> logs = cache.getLogs();
+        int i = 0;
+        while (i < logs.size() && markers.getChildCount() < 8) {
+            int marker = -1;
+            LogType lt = logs.get(i++).logType;
+            switch (lt) {
```

Comment:
IMHO this switch Statement is quite ugly. Why not adding a new parameter to the `LogType` enum?

Reply:
If the icon for a logtype was used in multiple places I'd fully agree, like this imho it's ok.

Do you think it'll be needed in additional places? Maybe we want to show the icon also in Logbook tab next to the status text?

Reply:
Currently it's the only place, but we should always try to make things reusable. Usage in the Logbook tab sounds like a not too bad idea BTW. 

But there are more reasons, what if e.g. there is the need to introduce a new log type, etc? I would consider it easier to maintain if everything to mapped at a single place (the Logtype enum). Cause especially in such complex open source projects as cgeo, nobody can really keep every code part in mind and a spreaded switch statement is quite hard to keep bi sync... That's exactly the reason why we extensively use and should use enums at many places in cgeo.

Reply:
ok, I'll move it to LogType.

#11684 opened for icons in logbook tab ;-)